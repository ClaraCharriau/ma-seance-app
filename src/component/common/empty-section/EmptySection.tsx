import { redirect } from 'react-router-dom';
import style from './EmptySection.module.css';

interface EmptySectionProps {
    itemType: 'theater' | 'movie';
}

const EmptySection = (props: EmptySectionProps) => {
    const { itemType } = props;

    return (
        <div className={style.emptySection}>
            {itemType === 'movie' ? (
                <p>Vous n'avez pas encore de films dans votre watchlist</p>
            ) : (
                <p>Vous n'avez pas encore de cinémas favoris</p>
            )}
            <button onClick={() => redirect('/search')} className={style.button}>
                Rechercher un {itemType === 'movie' ? 'film' : 'cinéma'}
            </button>
        </div>
    );
};

export default EmptySection;
