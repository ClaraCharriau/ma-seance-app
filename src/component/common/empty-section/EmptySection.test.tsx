import { fireEvent, render } from '@testing-library/react';
import { BrowserRouter } from 'react-router-dom';
import EmptySection from './EmptySection';

describe('Empty home section component', () => {
    it('renders Empty home section component for theater', () => {
        const component = render(
            <BrowserRouter>
                <EmptySection itemType={'theater'} />
            </BrowserRouter>
        );

        expect(component.container).toMatchSnapshot();
    });

    it('renders Empty home section component for movie', () => {
        const component = render(
            <BrowserRouter>
                <EmptySection itemType={'movie'} />
            </BrowserRouter>
        );

        expect(component.container).toMatchSnapshot();
    });

    it('should redirect on click', async () => {
        const component = render(
            <BrowserRouter>
                <EmptySection itemType={'movie'} />
            </BrowserRouter>
        );
        const button = await component.findByText('Rechercher un film');

        fireEvent.click(button);

        expect(component.container).toMatchSnapshot();
    });
});
