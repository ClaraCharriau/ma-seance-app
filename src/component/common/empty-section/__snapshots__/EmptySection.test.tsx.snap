// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Empty home section component renders Empty home section component for movie 1`] = `
<div>
  <div
    class="emptySection"
  >
    <p>
      Vous n'avez pas encore de films dans votre watchlist
    </p>
  </div>
</div>
`;

exports[`Empty home section component renders Empty home section component for theater 1`] = `
<div>
  <div
    class="emptySection"
  >
    <p>
      Vous n'avez pas encore de cin√©mas favoris
    </p>
  </div>
</div>
`;
